packages/cookie-consent-react/src/CookieConsent.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useEffect } from "react";
+import { usePreviousValue } from "@fremtind/jkl-react-hooks";
+import { Consent } from "./types";
+import { useCookieConsentState } from "./CookieConsentContext";
+import { CookieConsentModal } from "./CookieConsentModal";
+import { setConsentCookie } from "./cookieConsentUtils";
+
+interface Props {
+    blocking?: boolean;
+    onAccept?: (consent: Consent) => void;
+}
+
+export const CookieConsent = ({ blocking, onAccept }: Props) => {
+    const { dispatch, consent } = useCookieConsentState();
+    const prevConsent = usePreviousValue(consent);
+
+    useEffect(() => {
+        if (onAccept) {
+            // Prevent onAccept from being triggered every time state changes\
+            if (consent !== prevConsent) {
+                onAccept(consent);
+            }
+        }
+    }, [onAccept, consent, prevConsent]);
+
+    // Cookies are disabled in the browser
+    if (navigator && !navigator.cookieEnabled) {
+        if (onAccept) {
+            onAccept({
+                functional: "denied",
+                marketing: "denied",
+                statistics: "denied",
+            });
+        }
+
+        return null;
+    }
+
+    const handleAccept = (newConsent: Partial<Consent>) => {
+        // Filter out null values from the new consent
+        const newConsentWithoutNullValues: Record<string, string> = {};
+        for (const [consentName, consentValue] of Object.entries(newConsent)) {
```

Comment:
Jeg tror denne prosessen med fjerning av filtre kan simplifiseres til en linje. Var noen flere eksempler her: https://stackoverflow.com/questions/286141/remove-blank-attributes-from-an-object-in-javascript

Men likte denne one-linern
``` 
let newConsentWithoutNullValues = Object.fromEntries(Object.entries(newConsent).filter(([_, v]) => v != null));
```

Reply:
Kunne eventuelt laget en function som heter removeNullValues, er sikkert en funksjon som kan gjenbrukes andre steder også kan man evt lage tester for en slik funksjon også

Reply:
Den var snerten og fin!

Selv om det er et use-case som oppstår i ny og ne, ser jeg ikke behover for å opprette en "jkl-util"-aktig pakke for sånne funksjoner enda. Men skal ikke se bort i fra at denne hører hjemme i en sånn pakke over tid.